text/trace/0235-sampling-threshold-in-trace-state.md

```diff
@@ -0,0 +1,151 @@
+# Sampling Threshold Propagation in TraceState
+
+## Motivation
+
+Sampling is a broad topic; here it refers to the independent decisions made at points in a distributed tracing system of whether to collect a span or not. Multiple sampling decisions can be made before a span is finally consumed. When sampling is to be performed at multiple points in the process, the only way to reason about it effectively is to make sure that the sampling decisions are **consistent**.
+In this context, consistency means that a positive sampling decision made for a particular span with probability p1 implies a positive sampling decision for any span belonging to the same trace, if it is made with probability p2 >= p1.
+
+## Explanation
+
+The existing, experimental [specification for probability sampling using TraceState](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/tracestate-probability-sampling.md) is limited to powers-of-two probabilities, and is designed to work without making assumptions about TraceID randomness.
+This system can only achieve non-power-of-two sampling using interpolation between powers of two, which is unnecessarily restrictive.
+In existing sampling systems, sampling probabilities like 1%, 10%, and 75% are common, and it should be possible to express these without interpolation.
+There is also a need for consistent sampling in the collection path (outside of the head-sampling paths) and using inherent randomness in the traceID is a less-expensive solution than referencing a custom `r-value` from the tracestate in every span.
+This proposal introduces a new value with the key `th` as a replacement for the `p` value in the previous specification.
+The `p` value is limited to powers of two, while this proposal is not.
+This allows for the continued expression of randomness using `r-value` as specified there using the key `r`.
+To distinguish the cases, this proposal uses the key `rv`.
+
+In order to make consistent sampling decisions across the entire path of the trace, two values SHOULD be propagated with the trace:
+
+1. A _random_ (or pseudo-random) 56-bit value, called `R` below.
+2. A 56-bit trace _threshold_ as expressed in the TraceState, called `T` below. `T` represents the minimum threshold that was applied in all previous consistent sampling stages. If the current sampling stage applies a lower threshold than any stage before, it has to update (decrease) the threshold correspondingly.
+
+The system has the following invariant:
+
+`(T=0) OR ((R < T) = sampled flag)`
```

Comment:
I believe we need to state explicitly that `sampled` flag and the `T` value in the trace state document the last decision wrt the span.

Reply:
I am not sure what to change based on this feedback.

Reply:
@kentquirk Not sure if you're addressing me or Peter. In case me: IMO it's unclear what "the system" refers to. My suggested edit attempts to be more precise by using API spec terminology. My current understanding is that L45 can be viewed as a predicate that's always true for any `Span` in any process's memory following the creation via a Tracer.